tests/e2e/Services/Databases/DatabasesBase.php

```diff
@@ -1422,9 +1422,40 @@ public function testCreateIndexes(array $data): array
         return $data;
     }
 
+
     /**
-     * @depends testCreateIndexes
+     * @depends testCreateAttributes
      */
+    public function testGetIndexByKeyWithLengths(array $data): void
+    {
+        $databaseId = $data['databaseId'];
+        $collectionId = $data['moviesId'];
+
+        $create = $this->client->call(Client::METHOD_POST, "/databases/{$databaseId}/collections/{$collectionId}/indexes", [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ], [
+            'key' => 'lengthTestIndex',
+            'type' => 'key',
+            'attributes' => ['title','description'],
+            'lengths' => [128,200]
+        ]);
+
+        $this->assertEquals(202, $create['headers']['status-code']);
+
+        $index = $this->client->call(Client::METHOD_GET, "/databases/{$databaseId}/collections/{$collectionId}/indexes/lengthTestIndex", [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]);
+        $this->assertEquals(200, $index['headers']['status-code']);
+        $this->assertEquals('lengthTestIndex', $index['body']['key']);
+        $this->assertEquals([128,200], $index['body']['lengths']);
+    }
```

Comment:
Let's check some other cases as well:

- Lengths for array attributes get overridden correctly
- Lengths are assigned correctly when skipping some attributes (e.g. [null, null, 255] or [2 => 255])
- Count of lengths greater than count of attributes correctly throws 400
- Lengths too large throws 400 (total of all lengths should not exceed 768 for us, because we're using utf8mb4, 768 chars gives max 3072 bytes)
- Negative length values throws 400
- Any other failure cases you can think of

If you need to, add additional exception types + error messages to handle these cases